File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/schedule/ScheduleRepository.java
package yoyo.app.android.com.yoyoapp.trip.schedule;

import android.content.Context;
import androidx.lifecycle.MutableLiveData;
import yoyo.app.android.com.yoyoapp.DataModels.Schedule;
import yoyo.app.android.com.yoyoapp.trip.ApiService;

import java.util.ArrayList;

public class ScheduleRepository {

    private Context context;
    private ApiService apiService;
    private static ScheduleRepository instance;
    private MutableLiveData<ArrayList<Schedule>> scheduleMutabaleList;

    public ScheduleRepository(Context context) {
        this.context = context;
        apiService = new ApiService(context);
    }

    public static ScheduleRepository getInstance(Context context) {
        if (instance == null)
        {
            instance = new ScheduleRepository(context);
        }
        return instance;
    }

    public MutableLiveData<ArrayList<Schedule>> getSchedule(String tripId, long startDate, long endDate)
    {
        scheduleMutabaleList = new MutableLiveData<>();
        setScheduleList(tripId,startDate ,endDate);
        return scheduleMutabaleList;
    }

    private void setScheduleList(String tripId, long stardDate, long endDate) {
        apiService.getScheduleRequest(tripId,stardDate,endDate,schedules -> {
            scheduleMutabaleList.postValue(schedules);
        });
    }


}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/schedule/ScheduleRepository.kt
package yoyo.app.android.com.yoyoapp.trip.schedule

import android.content.Context
import androidx.lifecycle.MutableLiveData
import yoyo.app.android.com.yoyoapp.DataModels.Schedule
import yoyo.app.android.com.yoyoapp.trip.ApiService

import java.util.ArrayList

class ScheduleRepository(private val context: Context) {
    private val apiService: ApiService
    private var scheduleMutabaleList: MutableLiveData<ArrayList<Schedule>>? = null

    init {
        apiService = ApiService(context)
    }

    fun getSchedule(tripId: String, startDate: Long, endDate: Long): MutableLiveData<ArrayList<Schedule>> {
        scheduleMutabaleList = MutableLiveData()
        setScheduleList(tripId, startDate, endDate)
        return scheduleMutabaleList
    }

    private fun setScheduleList(tripId: String, stardDate: Long, endDate: Long) {
        apiService.getScheduleRequest(tripId, stardDate, endDate) { schedules ->
            scheduleMutabaleList!!.postValue(
                schedules
            )
        }
    }

    companion object {
        private var instance: ScheduleRepository? = null

        fun getInstance(context: Context): ScheduleRepository {
            if (instance == null) {
                instance = ScheduleRepository(context)
            }
            return instance
        }
    }


}
